spec/mongo/index/view_spec.rb

```diff
@@ -196,7 +196,70 @@
         expect(view.get('random_name')).to_not be_nil
       end
     end
-  end
+
+    context 'when providing an invalid partial index filter' do
+
+      let(:spec) do
+        { random:1 }
+      end
+
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: 5)
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: {"x": {"$asda": 3}})
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: {
+                                  "$and": [{"$and": [{"x": {"$lt": 2}},
+                                                     {"x": {"$gt": 0}}]},
+                                           {"x": {"$exists": true}}]})
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+    end
+
+    context 'when providing a valid partial index filter' do
+      let!(:result) do
+        view.create_one({"x": 1}, partial_filter_expression: {
+                                    "a": {"$lte": 1.5}})
+      end
+
+      it 'returns ok' do
+        expect(result).to be_successful
+      end
+
+      let!(:explain) do
+        authorized_collection.find(:x => 6, :a => 1).explain
+      end
+
+      it 'returns ok' do
+        expect (explain['queryPlanner']['winningPlan'].fetch(
+                   'inputStage', {})['indexName'] == "x_1")
+        expect (explain['queryPlanner']['winningPlan'].fetch(
+                   'inputStage', {})['isPartial'])
```

Comment:
I think you're using #fetch on the hash because 'inputStage' may or may not be a key. If you refer to a key in a Ruby hash that doesn't exist, it'll just return nil; it won't raise an exception.
So you can just do 

```ruby
expect (explain['queryPlanner']['winningPlan']['inputStage']).to eq('isPartial')
```
If explain['queryPlanner']['winningPlan'] doesn't have the key 'inputStage', the value of the expectation will be nil and it will be compared with the String 'isPartial'. Which would be false, failing the test.
You always want expect(ations) in rspec tests to actually inspect a value. Here you just have expect(..) and nothing else so you should add the ".to eq('isPartial')"